src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public Figure getRandomFigure() {
+        ColorSupplier colorSupplier = new ColorSupplier();
+        Random random = new Random();
+        int index = random.nextInt(Figures.values().length);
+        Figures figure = Figures.values()[index];
+
+        switch (figure.toString()) {
+            case "SQUARE":
+                return new Square(colorSupplier.getRandomColor(), random.nextDouble() * 100);
+            case "RECTANGLE":
+                return new Rectangle(colorSupplier.getRandomColor(),
+                        random.nextDouble() * 100, random.nextDouble() * 100);
+            case "RIGHT_TRIANGLE":
+                return new RightTriangle(colorSupplier.getRandomColor(), random.nextDouble() * 100,
+                        random.nextDouble() * 100, random.nextDouble() * 100);
+            case "CIRCLE":
+                return new Circle(colorSupplier.getRandomColor(), random.nextDouble() * 100);
+            case "ISOSCELES_TRAPEZOID":
+                return new IsoscelesTrapezoid(colorSupplier.getRandomColor(),
+                        random.nextDouble() * 100,
+                        random.nextDouble() * 100,
+                        random.nextDouble() * 100);
+            default:
+                return null;

```

Comment:
Returning `null` from a method is considered a bad practice. Consider handling the default case differently, such as by throwing an exception or returning a default figure.